### PR TITLE
Expose utils for expression LSP

### PR DESCRIPTION
**Pull request by adierkens**
## Release Notes
- Expose `parseExpression` from the expression parser
- Adds a `strict` flag to `parseExpression` that will (when disabled) attach any parsing errors to the top-level node instead of throwing

### CODE REVIEW

File: core/player/src/expressions/parser.ts [hunk 1]
```diff
@@ -899,37 +911,51 @@ export default function parseExpression(expr: string): ExpressionNode {
...
 
-  while (index < length) {
-    const chIndex = exprICode(index);
+  try {
+    while (index < length) {
+      const chIndex = exprICode(index);
+
+      // Expressions can be separated by semicolons, commas, or just inferred without any
+      // separators
+      if (chIndex === SEMCOL_CODE || chIndex === COMMA_CODE) {
```
Comment (KetanReddy) [inline core/player/src/expressions/parser.ts:920]: Would there be any worth to having the kind of characters that are ignored be configurable?

Comment (adierkens) [reply core/player/src/expressions/parser.ts:920]: That may make it pretty difficult to define a proper bnf grammar for expressions. But that might be worth considering as a future change

**Pull request merged**